src/plugins/Utilities/cdc_amp_t/JEventProcessor_cdc_amp_t.cc

```diff
@@ -0,0 +1,234 @@
+// $Id$
+//
+//    File: JEventProcessor_cdc_amp_t.cc
+// Created: Wed Aug 14 12:10:19 PM EDT 2024
+// Creator: njarvis (on Linux gluon01 5.14.0-427.22.1.el9_4.x86_64 x86_64)
+//
+
+#include "JEventProcessor_cdc_amp_t.h"
+using namespace jana;
+
+
+// Routine used to create our JEventProcessor
+#include <JANA/JApplication.h>
+#include <JANA/JFactory.h>
+
+#include "CDC/DCDCHit.h"
+#include "CDC/DCDCDigiHit.h"
+
+#include "DAQ/Df125CDCPulse.h"
+#include "DAQ/Df125Config.h"
+
+#include "TRIGGER/DTrigger.h"
+
+#include <TRACKING/DTrackTimeBased.h>
+#include <PID/DChargedTrackHypothesis.h>
+#include <PID/DChargedTrack.h>
+#include <TRACKING/DTrackFitter.h>
+#include "CDC/DCDCTrackHit.h"
+
+
+
+
+extern "C"{
+void InitPlugin(JApplication *app){
+	InitJANAPlugin(app);
+	app->AddProcessor(new JEventProcessor_cdc_amp_t());
+}
+} // "C"
+
+
+//------------------
+// JEventProcessor_cdc_amp_t (Constructor)
+//------------------
+JEventProcessor_cdc_amp_t::JEventProcessor_cdc_amp_t()
+{
+
+}
+
+//------------------
+// ~JEventProcessor_cdc_amp_t (Destructor)
+//------------------
+JEventProcessor_cdc_amp_t::~JEventProcessor_cdc_amp_t()
+{
+
+}
+
+//------------------
+// init
+//------------------
+jerror_t JEventProcessor_cdc_amp_t::init(void)
+{
+	// This is called once at program startup. 
+
+    amp_t = new TH2I("amp_t","Digihit pulse height - pedestal vs sample number;sample number;pulse height-pedestal", 200,0,200,205,0,4100);
```

Comment:
Both cdc_echo and cdc_amp_t have histograms with this name, so there is a conflict when running both plugins at the same time. It is a good practice to create a folder with the plugin name in the histogram root file.

Reply:
Okay.  They are the same histogram, both plugins have a very specific use case.  I guess I could delete the histogram from inside cdc_echo.

Reply:
Ok, the policy does not strictly apply to optional plugins.